ksql-engine/src/main/java/io/confluent/ksql/streams/KsqlGroupedFactory.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package io.confluent.ksql.streams;
+
+import static org.apache.kafka.streams.StreamsConfig.OPTIMIZE;
+import static org.apache.kafka.streams.StreamsConfig.TOPOLOGY_OPTIMIZATION;
+
+import io.confluent.ksql.util.KsqlConfig;
+import java.util.Map;
+import java.util.Objects;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.streams.kstream.Grouped;
+
+public class KsqlGroupedFactory implements GroupedFactory {
+  final KsqlConfig ksqlConfig;
```

Comment:
nit: can these not be private?

(And the class final?)

Reply:
Yeah. I'm surprised checkstyle wouldn't catch that.

Reply:
The whole point of these is so that I can mock them from tests, which is why I don't want to make them final